### PR TITLE
Add decoderbufs contrib

### PR DESCRIPTION
**Pull request by CyberDem0n**
* In addition to that switch to ubuntu 16.04. In order to avoid installing of some strange packages prepare and install bunch of fake packages which will cut off dependencies.

* bump commit version of pg_repack.

### CODE REVIEW

File: postgres-appliance/Dockerfile [hunk 1]
```diff
@@ -81,35 +102,36 @@ RUN export DEBIAN_FRONTEND=noninteractive \
     && curl -s -L https://github.com/zserge/jsmn/archive/$JSMN_COMMIT.tar.gz | tar xz \
     && rm -fr jsmn && mv jsmn-$JSMN_COMMIT jsmn \
     && make install \
-    && cd .. \
+    && cd ../.. \
 
     # Clean up
-    && apt-get purge -y ${BUILD_PACKAGES} \
+    && apt-get purge -y equivs fakeroot libfakeroot libmpc3 bsdmainutils libcc1-0 groff-base libunistring0 libmagic1 file ${BUILD_PACKAGES} \
```
Comment (CyberDem0n) [inline postgres-appliance/Dockerfile]: "fakeroot libfakeroot libmpc3 bsdmainutils libcc1-0 groff-base libunistring0 libmagic1 file" -- were pulled by equivs and somehow autoremove doesn't want to clean them up...

File: postgres-appliance/Dockerfile [hunk 2]
```diff
@@ -29,13 +29,30 @@ RUN export DEBIAN_FRONTEND=noninteractive \
...
 
     && apt-get update \
-    && apt-get install -y postgresql-common ${BUILD_PACKAGES} \
+    && apt-get install -y postgresql-common equivs gcc \
     # forbid creation of a main cluster when package is installed
     && sed -ri 's/#(create_main_cluster) .*$/\1 = false/' /etc/postgresql-common/createcluster.conf \
 
+    # Prepare and install some "fake" packages to avoid installing unnecessary dependencies
+    && mkdir tempdir && cd tempdir \
+    && for p in fontconfig-config libavcodec-ffmpeg56 libavdevice-ffmpeg56 libavformat-ffmpeg56 libavresample-ffmpeg2 \
```
Comment (CyberDem0n) [inline postgres-appliance/Dockerfile]: Fake packages to cut off dependencies.

File: postgres-appliance/Dockerfile [hunk 3]
```diff
@@ -56,6 +73,10 @@ RUN export DEBIAN_FRONTEND=noninteractive \
                 pgxn install $extension; \
             done \
 
+            && if [ "$version" != "9.3" ]; then \
+                make -C postgres-decoderbufs-$DECODEBUFS_VER clean install; \
```
Comment (CyberDem0n) [inline postgres-appliance/Dockerfile]: Some warnings...
```
src/decoderbufs.c: In function 'pg_decode_startup':
src/decoderbufs.c:116:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
   ListCell *option;                                                                                                                                                                                                                                                            
   ^                                                                                                                                                                                                                                                                            
src/decoderbufs.c: In function 'pg_decode_shutdown':                                                                                                                                                                                                                            
src/decoderbufs.c:161:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
   DecoderData *data = ctx->output_plugin_private;                                                                                                                                                                                                                              
   ^                                                                                                                                                                                                                                                                            
src/decoderbufs.c: In function 'set_datum_value':                                                                                                                                                                                                                               
src/decoderbufs.c:402:18: warning: initialization makes integer from pointer without a cast [-Wint-conversion]                                                                                                                                                                  
   Timestamp ts = NULL;                                                                                                                                                                                                                                                         
                  ^                                                                                                                                                                                                                                                             
src/decoderbufs.c:513:7: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
       Decoderbufs__Point dp = DECODERBUFS__POINT__INIT;                                                                                                                                                                                                                        
       ^                                                                                                                                                                                                                                                                        
src/decoderbufs.c:530:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
         int len = strlen(output);                                                                                                                                                                                                                                              
         ^                                                                                                                                                                                                                                                                      
src/decoderbufs.c: In function 'tuple_to_tuple_msg':                                                                                                                                                                                                                            
src/decoderbufs.c:571:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
     Decoderbufs__DatumMessage datum_msg = DECODERBUFS__DATUM_MESSAGE__INIT;                                                                                                                                                                                                    
     ^                                                                                                                                                                                                                                                                          
src/decoderbufs.c:583:27: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]                                                                                                                                                      
     datum_msg.column_name = attrName;                                                                                                                                                                                                                                          
                           ^                                                                                                                                                                                                                                                    
src/decoderbufs.c:592:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
     Oid typoutput;                                                                                                                                                                                                                                                             
     ^                                                                                                                                                                                                                                                                          
src/decoderbufs.c: In function 'pg_decode_change':                                                                                                                                                                                                                              
src/decoderbufs.c:619:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
   DecoderData *data;                                                                                                                                                                                                                                                           
   ^                                                                                                                                                                                                                                                                            
src/decoderbufs.c:656:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
         TupleDesc tupdesc = RelationGetDescr(relation);                                                                                                                                                                                                                        
         ^                                                                                                                                                                                                                                                                      
src/decoderbufs.c:671:11: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                  
           TupleDesc tupdesc = RelationGetDescr(relation);                                                                                                                                                                                                                      
           ^                                                                                                                                                                                                                                                                    
src/decoderbufs.c:680:11: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                  
           TupleDesc tupdesc = RelationGetDescr(relation);                                                                                                                                                                                                                      
           ^                                                                                                                                                                                                                                                                    
src/decoderbufs.c:696:9: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]                                                                                                                                                                   
         TupleDesc tupdesc = RelationGetDescr(relation);                                                                                                                                                                                                                        
         ^                                                                                                                                                                                                                                                                      
src/decoderbufs.c:625:70: warning: 'is_rel_non_selective' is used uninitialized in this function [-Wuninitialized]                                                                                                                                                              
   const char *selectiveInfo = is_rel_non_selective ? "non selective" : "selective";                                                                                                                                                                                            
```

**Comment by CyberDem0n** [comment]
Final size is still big, 793MB vs 677MB on ubuntu 14.04 :(

File: postgres-appliance/Dockerfile (hunk 3 above)
Comment (alexeyklyukin) [inline postgres-appliance/Dockerfile]: Yeah, the code was written by someone who probably thought C++ is just C without classes. Most of those are stupid declaration not preceding the code warnings, the pointer init in set_datum_value is harmless, as it is never used before being initialized again, and the is_rel_non_selective is only used in this context:
``
bool is_rel_non_selective;
 const char *selectiveInfo = is_rel_non_selective ? "non selective" : "selective";
``
and then initialized properly, so it's not a problem apart from the wrong debug message.

On the other hand, this indicates the code is rough (as it doesn't have any issues on github, so probably no-one used it yet), so we should tread lightly.

**Comment by alexeyklyukin** [comment]
👍

**Comment by alexeyklyukin** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**